spec/features/idv/doc_auth/document_capture_step_spec.rb

```diff
@@ -314,6 +320,27 @@
         expect(page).to have_current_path(next_step)
       end
     end
+
+    context 'when using async uploads', :js do
+      let(:document_capture_async_uploads_enabled) { true }
+      let(:doc_auth_enable_presigned_s3_urls) { true }
+
+      it 'proceeds to the next page with valid info' do
+        mock_document_capture_result
+        attach_images(liveness_enabled: false)
+        form = page.find('#document-capture-form')
+        front_url = form['data-front-image-upload-url']
+        back_url = form['data-back-image-upload-url']
+        click_on 'Submit'
+
+        expect(page).to have_current_path(next_step, wait: 20)
+        Capybara.current_driver = :rack_test # ChromeDriver doesn't support `page.status_code`
+        visit front_url
+        expect(page).to have_http_status(200)
+        visit back_url
+        expect(page).to have_http_status(200)
```

Comment:
Ah! We can issue direct requests, [here's an example in the OIDC tests](https://github.com/18F/identity-idp/blob/master/spec/features/openid_connect/openid_connect_spec.rb#L169-L171)

I just realized the body check below will fail due to the encryption....unsure if it's worth trying to extract the key and IV values to decrypt....but it is an option!

```suggestion
        page.driver.get front_url
        expect(page).to have_http_status(200)
        expect(page.body).to eq(File.read('app/assets/images/logo.png')) # TODO: decrypt?

        page.driver.get back_url
        expect(page).to have_http_status(200)
        expect(page.body).to eq(File.read('app/assets/images/logo.png')) # TODO: decrypt?
```

Reply:
Oh maybe we do need the driver setting there for the status code, I see what it's doing now, will update the patch

Reply:
Interestingly, I started from `page.driver.get`, which also doesn't exist in the ChromeDriver... but the fact that we eventually arrived at switching drivers mid-test means it could be used now.

I'll try to poke at doing a decrypt + assert. I agree that'd be great to have.

At the same time, I'm not quite sure what I'm seeing with the build failures, so will have to work through that as well 🤔

Reply:
My guess with the with build failures.... maybe we just apply the hostname patch to these specs and leave the others as-is? There's a lot of domain name shenanigans

Reply:
> My guess with the with build failures.... maybe we just apply the hostname patch to these specs and leave the others as-is? There's a lot of domain name shenanigans

I thought so as well, though it doesn't seem at first glance those tests use the JavaScript driver, and those tests pass locally for me with a fresh `application.yml`. Interestingly, I'd seen the same errors _before_ resetting `application.yml` to default, but I would think the CircleCI build should always be running fresh.

Gonna try a rebase for good measure, but as a fallback will try to constraining patching to these tests specifically.

Reply:
It ain't the prettiest, but it's passing!

ddc7e0f

(I might give it another look over on Monday when I'm of a fresher mind)

Reply:
> I might give it another look over on Monday when I'm of a fresher mind

I'll plan to leave it as-is. There's a bit of duplication, but I don't really think it's worth the effort to refactor.